feat(clerk-js,clerk-react,types): Introduce `UserButton` custom menu items (#3784)
clerk/javascript: .changeset/smooth-worms-retire.md

---
"@clerk/clerk-js": minor
"@clerk/clerk-react": minor
"@clerk/types": minor
---

> Extended the UserButton component to support custom menu items:

Reviewer: ```suggestion
Extended the `UserButton` component to support custom menu items.
```

Author: Done

> Extended the `UserButton` component to support custom menu items:

Reviewer 2: Introduce support for custom menu items in `<UserButton/>`.

Author: Done

> - Introduced `UserButton.MenuItems` as a child component to wrap custom menu items

Reviewer 2: ```suggestion
- Use `<UserButton.MenuItems>` as a child component to wrap custom menu items.
```

Author: Done

> - Introduced `UserButton.Link` for custom link items

Reviewer 2: ```suggestion
- Use `<UserButton.Link/>` for creating external or internal links.
```

> - Introduced `UserButton.Action` for custom action items

Reviewer 2: ```suggestion
- Use `<UserButton.Action/>` for opening a specific custom page of "UserProfile" or to trigger your own custom logic via `onClick`.
```

Author: Done

> - Implemented support for reordering existing items like `manageAccount` and `signOut`

Reviewer 2: ```suggestion
- If needed, reorder existing items like `manageAccount` and `signOut`.
```

Author: Done

New usage example:

> ```jsx

Reviewer 3: You're missing the closing backticks for this code-block here

Author: Fixed

<UserButton>
  <UserButton.MenuItems>
    <UserButton.Link label='Terms' labelIcon={<Icon />} href='/terms' />

>     <UserButton.Action label='Help' labelIcon={<Icon />} open='help' /> // Opens help page in UserProfileModal

Reviewer 2: ```suggestion
    <UserButton.Action label='Help' labelIcon={<Icon />} open='help' /> // Navigate to `/help` page when UserProfile opens as a modal. (Requires a custom page to have been set in `/help`)
```

Author: Done
